subsys/net/lib/http/http_server_http1.c

```diff
@@ -256,6 +257,128 @@ static int dynamic_post_req(struct http_resource_detail_dynamic *dynamic_detail,
 	return 0;
 }
 
+#if defined(CONFIG_FILE_SYSTEM)
+int handle_http1_static_fs_resource(struct http_resource_detail_static_fs *static_fs_detail,
+				    struct http_client_ctx *client)
+{
+#define RESPONSE_TEMPLATE_STATIC_FS                                                                \
+	"HTTP/1.1 200 OK\r\n"                                                                      \
+	"Content-Type: %s%s\r\n\r\n"
+#define CONTENT_ENCODING_GZIP "\r\nContent-Encoding: gzip"
+
+	bool gzipped = false;
+	int len;
+	int remaining;
+	int ret;
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	char fname[256];
```

Comment:
Do we really need to have this long filename here, it causes lot of pressure to the used stack size.

Reply:
`fname` needs to consume `fs_path` and and the `url_buffer`. The `url_buffer` is 256 bytes long, to be safe we would need 256 + `strlen(fs_path)`. I changed the size to `HTTP_SERVER_MAX_URL_LENGTH`, so the user can reduce the buffer size when he runs short on stack size.